### ISSUE TITLE
TypeError: Unrecognised typecode: 5

### ISSUE THREAD

**Issue by frannyfx** [amqp/rhea#370]
When receiving messages from AMQP.Net Lite, rhea seemingly is using the wrong offsets in the buffer to deduce the typecode of properties from, as it gets erroneous values such as 5 or 1, and throws "TypeError: Unrecognised typecode: 5".

The weird thing about this is that rhea's predecessor, the amqp10 node library, handles the same exact messages just fine.
Any ideas?

**Comment by grs** [comment]
Can you get a wireshark trace of the communication? Or failing that logging for a simple reproducer (run with DEBUG=rhea*)?

**Comment by frannyfx** [comment]
Here is the buffer of the message being received

```
  rhea:io [connection-1] read 466 bytes +12s
  rhea:io [connection-1] got frame of size 466 +0ms
  rhea:raw [connection-1] RECV: 466 000001d202000000005314c008054343a001004342005370c002014100537700a33153696e6172612e546c632e494d502e456d61696c53656e6465722e496e746572666163652e456d61696c52657175657374d00000017a000000045401405401d10000016c00000004a11150617373776f726452657365744c696e6bb10000013568747470733a2f2f6c6f63616c686f73743a34343337342f466f72676f7450617373776f72643f69643d4366444a384933356c6e6b30424135507245435944412d5a494832515041303845794544684852704c56492d724753356e6c314445415450503036596e564170375878447a69436b463174344f4f69784244654970527078454b59695133443854422d577139746f6f6c53696e496d7450397a4c4f4838777755446f6a76556f5a33534e4b51266f74703d556c364d777241486b4a4f4c4e38615734464f4f57737a57636b344741646c727173667164484878547772686e744c544b48417352574146725953656367614137636136725737377546416b33766a6445446669756b6f387171445469775968484951475a4347414862666f4e4a575a4566766f36727552686b6b7058374275a10f52657365744c696e6b457870697279a10830303a31303a3030 +0ms
  rhea:frames [connection-1]:0 <- transfer#14 {"delivery_tag":{"type":"Buffer","data":[0]}} <Buffer 00 53 70 c0 02 01 41 00 53 77 00 a3 31 53 69 6e 61 72 61 2e 54 6c 63 2e 49 4d 50 2e 45 6d 61 69 6c 53 65 6e 64 65 72 2e 49 6e 74 65 72 66 61 63 65 2e ... 395 more bytes> +1ms
  rhea:message decoding section: Typed { type: TypeDesc { name: 'List8', typecode: 192, width: 1, category: 3, create: [Function (anonymous)] { typecode: 192 } }, value: [ Typed { type: [TypeDesc], value: true } ], descriptor: Typed { type: TypeDesc { name: 'SmallUlong', typecode: 83, width: 1, category: 1, read: [Function: read], write: [Function: write], create: [Function] }, value: 112 } } of type: Typed { type: TypeDesc { name: 'SmallUlong', typecode: 83, width: 1, category: 1, read: [Function: read], write: [Function: write], create: [Function (anonymous)] { typecode: 83 } }, value: 112 } +1ms
  rhea:events [connection-1] Connection got event: error +2ms
  rhea:events [f6a77377-e6c0-094c-a78e-2db93b0fac2c] Container got event: error +0ms
```

The exception being thrown is `TypeError: Unrecognised typecode: 1`.

Here is the output of the amqp10 library receiving a very similar message:

```
DescribedType {
  descriptor: '[[ custom message descriptor ]]',
  value: [
    1,
    null,
    1,
    {
      PasswordResetLink: 'https://localhost:44374/ForgotPassword?id=CfDJ8I35lnk0BA5PrECYDA-ZIH0OObTE8xyCusQDdHDC6C5i4y5vzefCXZHQz3rWEMDR6DvyTWCRV-GLAs_f89u1T7C9oc6VrTQ9v20oWj_IP4UOxCRb-Z7qPPzF5mGND9DTUg&otp=k5itVSikqd34GvSTQzbNPLtD3eYp6WRjsfHjKdxe8E6ZOxaa9dLs7mVIzkyHb5eNInZ0Mu1tFHAKt2NmmJTOF3qurLpyz2FxEzulCXDNauotZCv4cafeLke98n7FUpVN',
      ResetLinkExpiry: '00:10:00'
    }
  ]
}
```

**Comment by grs** [comment]
This was a bug in rhea. Should be fixed in 2.0.6
